text/0000-use-nested-groups.md

```diff
@@ -0,0 +1,177 @@
+- Feature Name: use_nested_groups
+- Start Date: 2017-08-25
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Permit nested `{}` groups in imports.  
+Permit `*` in `{}` groups in imports.
+
+```rust
+use syntax::{
+    tokenstream::TokenTree, // >1 segments
+    ext::base::{ExtCtxt, MacResult, DummyResult, MacEager}, // nested braces
+    ext::build::AstBuilder,
+    ext::quote::rt::Span,
+};
+
+use syntax::ast::{self, *}; // * in braces
+
+use rustc::mir::{*, transform::{MirPass, MirSource}}; // both * and nested braces
+```
+
+# Motivation
+[motivation]: #motivation
+
+The motivation is ergonomics.
+Prefixes are often shared among imports, especially if many imports
+import names from the same crate. With this nested grouping it's more often
+possible to merge common import prefixes and write them once instead of writing
+them multiple times.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Several `use` items with common prefix can be merged into one `use` item,
+in which the prefix is written once and all the suffixes are listed inside
+curly braces `{}`.  
+All kinds of suffixes can be listed inside curly braces, including globs `*` and
+"subtrees" with their own curly braces.
+
+```rust
+// BEFORE
+use syntax::tokenstream::TokenTree;
+use syntax::ext::base::{ExtCtxt, MacResult, DummyResult, MacEager};
+use syntax::ext::build::AstBuilder,
+use syntax::ext::quote::rt::Span,
+
+use syntax::ast;
+use syntax::ast::*;
+
+use rustc::mir::*;
+use rustc::transform::{MirPass, MirSource};
```

Comment:
Small correction: here the path should be `rustc::mir::transform::{MirPass, MirSource};`.